src/components/Receiving/ItemDetails.js

```diff
@@ -6,62 +6,160 @@ import { get } from 'lodash';
 
 import {
   Button,
+  Checkbox,
+  KeyValue,
   Modal,
+  MultiColumnList,
   Row,
   Select,
   TextField,
-  MultiColumnList,
 } from '@folio/stripes/components';
 
+import getLocationsForSelect from '../Utils/getLocationsForSelect';
+import css from './ItemDetails.css';
+
+const ITEM_STATUS = {
+  received: 'Received',
+};
+
 class ItemDetails extends Component {
   static propTypes = {
     close: PropTypes.func.isRequired,
     itemList: PropTypes.arrayOf(PropTypes.object).isRequired,
+    locations: PropTypes.arrayOf(PropTypes.object).isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      allChecked: false,
+      checkedItems: [],
+    };
+  }
+
+  getFooter = (close) => (
+    <Row end="xs">
+      <Button
+        onClick={close}
+      >
+        <FormattedMessage id="ui-orders.receiving.cancelBtn" />
+      </Button>
+      <Button
+        buttonStyle="primary"
+        disabled
+      >
+        <FormattedMessage id="ui-orders.receiving.previousBtn" />
+      </Button>
+      <Button
+        buttonStyle="primary"
+        disabled={!this.state.checkedItems.length}
+      >
+        <FormattedMessage id="ui-orders.receiving.nextBtn" />
+      </Button>
+    </Row>
+  );
+
+  toggleAll = (itemList) => {
+    if (!this.state.allChecked) {
+      this.setState(({ allChecked }) => ({
+        allChecked: !allChecked,
+        checkedItems: itemList,
+      }));
+    } else {
+      this.setState(({ allChecked }) => ({
+        allChecked: !allChecked,
+        checkedItems: [],
+      }));
+    }
+  }
+
+  toggleItem = (item) => {
+    const isItemChecked = this.state.checkedItems.filter(el => el.id === item.id).length;
+
+    if (isItemChecked) {
+      const checkedItemList = this.state.checkedItems.filter(el => el.id !== item.id);
+
+      this.setState({ checkedItems: [...checkedItemList] });
```

Comment:
you're changing `checkedItems` in state with values from `checkedItems` in current state. This usually done in callback of `this.setState()` https://reactjs.org/docs/react-component.html#setstate